test/e2e/apps/statefulset.go

```diff
@@ -734,10 +753,18 @@ var _ = SIGDescribe("StatefulSet", func() {
 
 			var initialStatefulPodUID types.UID
 			By("Waiting until stateful pod " + statefulPodName + " will be recreated and deleted at least once in namespace " + f.Namespace.Name)
-			w, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Watch(metav1.SingleObject(metav1.ObjectMeta{Name: statefulPodName}))
-			framework.ExpectNoError(err)
+			watchFunc := func(sinceResourceVersion string) (watch.Interface, error) {
```

Comment:
This is a ton more boilerplate that callers are likely to get wrong.  While I don't know that I want to fix it here, I'm verye concerned about how likely someone is to get these methods wrong as a consequence of changing their code.

This seems like something that should be a generated method in the clients package that takes a `metav1.ListOptions` and does the right thing.  `podclientset.WatchSelector(opts metav1.ListOptions) func(rv string) (watch.Interface, error)`

Reply:
Yes, there is a bit more you need to write, I wouldn't call *a ton* since it's really just 2 lines - wrapping it in a lambda and setting RV option. There is a potential to get it wrong as always, but I've fixed it throughout the kube code base so given that people usually copy the pattern, they should be good and as always having a test for their code will make them aware if it works or not from the start.

Enhancing that experience with generated code could be nice follow up ;)